File: in_Python/0929 Unique Email Addresses.py
class Solution(object):
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        temp = []
        for w in emails:
            t = w.split('@')
            s1 = t[0]
            s2 = t[1]
            new_s1 = ""
            for i in range(len(s1)):
                if s1[i] == '.':
                    continue
                elif s1[i] == '+':
                    break
                else:
                    new_s1 += s1[i]
            new_s = new_s1 + '@' + s2
            if new_s not in temp:
                temp.append(new_s)
        return len(temp)
        

File: in_Java/0929 Unique Email Addresses.java
class Solution {
    public int numUniqueEmails(String[] emails) {
        HashSet<String> hs = new HashSet<>();
        for(String s : emails){
            String[] ss = s.split("@");
            String name = ss[0];
            StringBuilder sb = new StringBuilder();
            for(int i = 0; i < name.length(); i++){
                String t = name.substring(i,i+1);
                if(t.equals("+")) break;
                if(!t.equals(".")) sb.append(t);
            }
            name = sb.toString();
            hs.add(name + "@" + ss[1]);
        }
        return hs.size();
    }
}